### ISSUE TITLE
Padding with opset 11 fails with model exported from PyTorch

### ISSUE THREAD

**Issue by DiegoTorresSED** [onnx/onnx-tensorrt#452]
Hi there, 
I am getting problems when trying to parser an ONNX model exported from PyTorch that uses a padding layer.
I have checked that in  #408  it has been added the padding support for opset 11 (moved "pads" from attributes to inputs, in this case, input at index 1 according to [https://github.com/onnx/onnx/blob/master/docs/Operators.md#Pad](url))
The case is with the model exported from PyTorch, parsing padding expects that "pads" were located at index 1 of inputs, but assert it is launched due to index 1 is a tensor instead of weights as expected in the code part (in DEFINE_BUILTIN_OP_IMPORTER(Pad)):
```cpp
ASSERT(inputs.at(1).is_weights(), ErrorCode::kUNSUPPORTED_NODE);
weightsToVector(inputs.at(1).weights(), &onnxPadding);
```

I have debugged this part and as I have said inputs.at(1) is a tensor with dimensions (-1) and inputs.at(2) is weights  with count 1 and values does not contain the expected padding values. 
Any help with this? Is there other case that has to be implemented in the OP importer? Could be a problem exporting from PyTorh?

This is my environment:
Anaconda 2020.02 (Python 3.7.6 64-bit)
PyTorh 1.4.0
TensorRT 7.0.0.11
onnx-tensorrt (compiled from master branch)

Padding layer from PyTorch is added using:

```python
class ConvZeroPadLeakyReLu(torch.nn.Module):
    def __init__(self):
        super().__init__()
        # Core
        self.core = Sequential()

        # Init layer expecting (3, 416, 416) input tensor
        self.core.add_module(
            'Conv2dBeforeZeroPad', 
            Conv2d(3, 32, kernel_size=3, stride=1, padding=(3-1) // 2))
        self.core.add_module(
            'LeakyReLUBeforeZeroPad', 
            LeakyReLU(negative_slope=0.1))
        self.core.add_module(
            'ZeroPad2d', 
            ZeroPad2d((1,0,1,0)))
        self.core.add_module(
            'Conv2dAfterZeroPad', 
            Conv2d(32, 64, kernel_size=3, stride=2, padding=0))
        self.core.add_module(
            'LeakyReLUAfterZeroPad', 
            LeakyReLU(negative_slope=0.1))
    
    def forward(self, x):
        y = self.core(x)
        return y
```

This is the output using `onnx2trt -vv ConvZeroPadLeakyReLu.onnx -o ConvZeroPadLeakyReLu.engine`

```
----------------------------------------------------------------
Input filename:   .\ConvZeroPadLeakyReLu.onnx
ONNX IR version:  0.0.4
Opset version:    0
Producer name:    pytorch
Producer version:
Domain:
Model version:    0
Doc string:
----------------------------------------------------------------
Writing ONNX model (without weights) as text to .\ConvZeroPadLeakyReLu.txt
Parsing model
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::GridAnchor_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::NMS_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Reorg_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Region_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Clip_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::LReLU_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::PriorBox_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Normalize_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::RPROI_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::BatchedNMS_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::FlattenConcat_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::CropAndResize
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::DetectionLayer_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Proposal
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::ProposalLayer_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::PyramidROIAlign_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::ResizeNearest_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::Split
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::SpecialSlice_TRT
Plugin creator registration succeeded - ONNXTRT_NAMESPACE::InstanceNormalization_TRT
ModelImporter.cpp:204: Adding network input: input_tensor with dtype: float32, dimensions: (1, 3, 416, 416)
ImporterContext.hpp:102: Registering tensor: input_tensor for ONNX tensor: input_tensor
ModelImporter.cpp:90: Importing initializer: 33
onnx2trt_utils.cpp:235: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
ModelImporter.cpp:90: Importing initializer: core.Conv2dAfterZeroPad.bias
ModelImporter.cpp:90: Importing initializer: core.Conv2dAfterZeroPad.weight
ModelImporter.cpp:90: Importing initializer: core.Conv2dBeforeZeroPad.bias
ModelImporter.cpp:90: Importing initializer: core.Conv2dBeforeZeroPad.weight
ModelImporter.cpp:107: Parsing node:  [Conv]
ModelImporter.cpp:123: Searching for input: input_tensor
ModelImporter.cpp:123: Searching for input: core.Conv2dBeforeZeroPad.weight
ModelImporter.cpp:123: Searching for input: core.Conv2dBeforeZeroPad.bias
ModelImporter.cpp:129:  [Conv] inputs: [input_tensor -> (1, 3, 416, 416)], [core.Conv2dBeforeZeroPad.weight -> (32, 3, 3, 3)], [core.Conv2dBeforeZeroPad.bias -> (32)],
builtin_op_importers.cpp:444: Convolution input dimensions: (1, 3, 416, 416)
builtin_op_importers.cpp:526: Using kernel: (3, 3), strides: (1, 1), padding: (1, 1), dilations: (1, 1), numOutputs: 32
builtin_op_importers.cpp:527: Convolution output dimensions: (1, 32, 416, 416)
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 0) [Convolution] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 5 for ONNX tensor: 5
ModelImporter.cpp:181:  [Conv] outputs: [5 -> (1, 32, 416, 416)],
ModelImporter.cpp:107: Parsing node:  [LeakyRelu]
ModelImporter.cpp:123: Searching for input: 5
ModelImporter.cpp:129:  [LeakyRelu] inputs: [5 -> (1, 32, 416, 416)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 1) [Activation] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 6 for ONNX tensor: 6
ModelImporter.cpp:181:  [LeakyRelu] outputs: [6 -> (1, 32, 416, 416)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [7 -> (4)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [8 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Shape]
ModelImporter.cpp:123: Searching for input: 7
ModelImporter.cpp:129:  [Shape] inputs: [7 -> (4)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 2) [Constant] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 9 for ONNX tensor: 9
ModelImporter.cpp:181:  [Shape] outputs: [9 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Gather]
ModelImporter.cpp:123: Searching for input: 9
ModelImporter.cpp:123: Searching for input: 8
ModelImporter.cpp:129:  [Gather] inputs: [9 -> (1)], [8 -> (1)],
builtin_op_importers.cpp:1015: Using Gather axis: 0
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 4) [Constant] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 10 for ONNX tensor: 10
ModelImporter.cpp:181:  [Gather] outputs: [10 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [11 -> ()],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [12 -> ()],
ModelImporter.cpp:107: Parsing node:  [Mul]
ModelImporter.cpp:123: Searching for input: 11
ModelImporter.cpp:123: Searching for input: 12
ModelImporter.cpp:129:  [Mul] inputs: [11 -> ()], [12 -> ()],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 6) [Constant] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 13 for ONNX tensor: 13
ModelImporter.cpp:181:  [Mul] outputs: [13 -> ()],
ModelImporter.cpp:107: Parsing node:  [Sub]
ModelImporter.cpp:123: Searching for input: 13
ModelImporter.cpp:123: Searching for input: 10
ModelImporter.cpp:129:  [Sub] inputs: [13 -> ()], [10 -> (1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 9) [Shuffle] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 14 for ONNX tensor: 14
ModelImporter.cpp:181:  [Sub] outputs: [14 -> (1)],
ModelImporter.cpp:107: Parsing node:  [ConstantOfShape]
ModelImporter.cpp:123: Searching for input: 14
ModelImporter.cpp:129:  [ConstantOfShape] inputs: [14 -> (1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 11) [Constant] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 16 for ONNX tensor: 16
ModelImporter.cpp:181:  [ConstantOfShape] outputs: [16 -> (-1)],
ModelImporter.cpp:107: Parsing node:  [Concat]
ModelImporter.cpp:123: Searching for input: 33
ModelImporter.cpp:123: Searching for input: 16
ModelImporter.cpp:129:  [Concat] inputs: [33 -> (4)], [16 -> (-1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 15) [Constant] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 17 for ONNX tensor: 17
ModelImporter.cpp:181:  [Concat] outputs: [17 -> (-1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [18 -> (2)],
ModelImporter.cpp:107: Parsing node:  [Reshape]
ModelImporter.cpp:123: Searching for input: 17
ModelImporter.cpp:123: Searching for input: 18
ModelImporter.cpp:129:  [Reshape] inputs: [17 -> (-1)], [18 -> (2)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 17) [Shuffle] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 19 for ONNX tensor: 19
ModelImporter.cpp:181:  [Reshape] outputs: [19 -> (-1, 2)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [20 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [21 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
onnx2trt_utils.cpp:251: Weight at index 0: -9223372036854775807 is out of range. Clamping to: -2147483648
onnx2trt_utils.cpp:261: One or more weights outside the range of INT32 was clamped
ModelImporter.cpp:181:  [Constant] outputs: [22 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [23 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Slice]
ModelImporter.cpp:123: Searching for input: 19
ModelImporter.cpp:123: Searching for input: 21
ModelImporter.cpp:123: Searching for input: 22
ModelImporter.cpp:123: Searching for input: 20
ModelImporter.cpp:123: Searching for input: 23
ModelImporter.cpp:129:  [Slice] inputs: [19 -> (-1, 2)], [21 -> (1)], [22 -> (1)], [20 -> (1)], [23 -> (1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 18) [Shape] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 24 for ONNX tensor: 24
ModelImporter.cpp:181:  [Slice] outputs: [24 -> (-1, -1)],
ModelImporter.cpp:107: Parsing node:  [Transpose]
ModelImporter.cpp:123: Searching for input: 24
ModelImporter.cpp:129:  [Transpose] inputs: [24 -> (-1, -1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 47) [Shuffle] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 25 for ONNX tensor: 25
ModelImporter.cpp:181:  [Transpose] outputs: [25 -> (-1, -1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [26 -> (1)],
ModelImporter.cpp:107: Parsing node:  [Reshape]
ModelImporter.cpp:123: Searching for input: 25
ModelImporter.cpp:123: Searching for input: 26
ModelImporter.cpp:129:  [Reshape] inputs: [25 -> (-1, -1)], [26 -> (1)],
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 48) [Shuffle] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 27 for ONNX tensor: 27
ModelImporter.cpp:181:  [Reshape] outputs: [27 -> (-1)],
ModelImporter.cpp:107: Parsing node:  [Cast]
ModelImporter.cpp:123: Searching for input: 27
ModelImporter.cpp:129:  [Cast] inputs: [27 -> (-1)],
builtin_op_importers.cpp:314: Casting to type: int32
ImporterContext.hpp:127: Registering layer: (Unnamed Layer* 49) [Identity] for ONNX node:
ImporterContext.hpp:102: Registering tensor: 28 for ONNX tensor: 28
ModelImporter.cpp:181:  [Cast] outputs: [28 -> (-1)],
ModelImporter.cpp:107: Parsing node:  [Constant]
ModelImporter.cpp:129:  [Constant] inputs:
ModelImporter.cpp:181:  [Constant] outputs: [29 -> ()],
ModelImporter.cpp:107: Parsing node:  [Pad]
ModelImporter.cpp:123: Searching for input: 6
ModelImporter.cpp:123: Searching for input: 28
ModelImporter.cpp:123: Searching for input: 29
ModelImporter.cpp:129:  [Pad] inputs: [6 -> (1, 32, 416, 416)], [28 -> (-1)], [29 -> ()],
```

**Comment by SBodenstedt** [comment]
I am encountering the exact same issue, did you ever find a solution?

**Comment by kevinch-nv** [comment]
@DiegoTorresSED @SBodenstedt unfortunately TensorRT only supports constant padding, which means that the padding values provided by the first input of the pad operator must be an initializer (or weights).

You can try "simplifying" your model with the following tool: https://github.com/daquexian/onnx-simplifier. It looks like in your Pytorch code you are doing a constant pad, so we should be able to constant-fold the pad. 

Try out the the tool with your model, and let me know if it worked for you

**Comment by opeide** [comment]
Exact same issue when using pytorch (1.8) onnx export and tensorrt (7.2) onnx parsing. Not sure which who is at fault here, but there is clearly a serious bug when you can't even export then import a simple padding function. How has this not been fixed or seriously addressed yet?

**Comment by kevinch-nv** [comment]
This has been addressed in TensorRT 8.2 where we've added support for generic padding.

Closing this issue for now - if there are still problems with TRT 8.2 please open a new issue.
